pkg/gce-pd-csi-driver/controller.go

```diff
@@ -382,7 +384,12 @@ func (gceCS *GCEControllerServer) ControllerUnpublishVolume(ctx context.Context,
 	}
 	instance, err := gceCS.CloudProvider.GetInstanceOrError(ctx, instanceZone, instanceName)
 	if err != nil {
-		return nil, err
+		if gce.IsGCENotFoundError(err) {
+			// Node not existing on GCE means that disk has been detached
+			klog.Warningf("Treating volume %v as unpublished because node %v could not be found", volKey.String(), instanceName)
+			return &csi.ControllerUnpublishVolumeResponse{}, nil
```

Comment:
Discussed this with @jsafrane this morning.  Current CSI spec clearly says to return NotFound error in this case.  Until CSI spec changes, we should probably follow it.

Reply:
Sorry I'm behind the times. This was discussed in the CSI spec meeting this morning and conclusion is we should allow plugins to return success if it knows volume is definitely detached.